backend/src/main/java/gov/cdc/usds/simplereport/service/DeviceTypeService.java

```diff
@@ -136,37 +94,30 @@ public DeviceType updateDeviceType(UpdateDeviceType updateDevice) {
               .collect(Collectors.toList());
 
       List<DeviceSpecimenType> exitingDeviceSpecimenTypes =
-          _deviceSpecimenRepo.findAllByDeviceType(device);
+          deviceSpecimenTypeRepository.findAllByDeviceType(device);
 
       // delete old ones
       ArrayList<DeviceSpecimenType> toBeDeletedDeviceSpecimenTypes =
           new ArrayList<>(exitingDeviceSpecimenTypes);
       toBeDeletedDeviceSpecimenTypes.removeAll(newDeviceSpecimenTypes);
-
-      // Null out facilities' default device specimen if it was deleted
-      List<Facility> facilitiesToRemoveDefaultDeviceSpecimen =
-          _facilityRepo.findAllByDefaultDeviceSpecimenIn(toBeDeletedDeviceSpecimenTypes);
-
-      facilitiesToRemoveDefaultDeviceSpecimen.forEach(Facility::removeDefaultDeviceSpecimen);
```

Comment:
is it possible that if we update a device type and remove a specimen type, any facilities that have that device and specimen type combo as their default could reflect an incorrect pair?
e.g.
facility A has valid defaults of device A and specimen type 1
we update the specimen types for device A to remove specimen type 1 as a valid specimen type
we add a patient to queue in that facility

won't that test order have an invalid device type/specimen type pair?

Reply:
yes, but why would you remove a specimen type from a device? 

- devices don't just stop supporting a specimen type.
- if the case is an admin error when entering the device specimen type / device pairing then the issue is with the source of truth, which hopefully the LIVD api integration will eliminate.

The previous checks were added to remove references invalid to deleted (hard deleted) `DeviceSpecimenTypes` because having a dead reference causes a facility wide crash in the backend.

The UI also handles the case, per work done in preparation for this PR https://github.com/CDCgov/prime-simplereport/pull/4728
![image](https://user-images.githubusercontent.com/4952042/214407813-12c7d970-e74d-4ef8-bb2f-5c9b9a1e53de.png)

Reply:
> devices don't just stop supporting a specimen type.
even though that's logical, I don't know if I agree that this is something we can confidently say since we don't own that data set. unless this is a guarantee of the data set we're using

thanks for pointing out that the UI handles that - I hadn't checked what happens. in that case it lgtm

Reply:
yeah that's true, there might be a correction from the LIVD table, but anyway, we're equipped to handle it. Thanks for the great feedback as always